File: NYCardCreator.java
// NYCardCreator.java
import java.awt.*;
import java.awt.image.BufferedImage;
import java.io.File;
import javax.imageio.ImageIO;
import java.util.*;

public class NYCardCreator {
    private static final Map<Integer, Color[]> TEMPLATES = new HashMap<>();
    static {
        TEMPLATES.put(1, new Color[]{Color.WHITE, Color.BLACK});
        TEMPLATES.put(2, new Color[]{new Color(0, 100, 150), new Color(255, 215, 0)});
        TEMPLATES.put(3, new Color[]{new Color(200, 50, 50), Color.WHITE});
    }

    public static Color hexToColor(String hex) {
        if (hex.startsWith("#")) hex = hex.substring(1);
        return new Color(Integer.parseInt(hex.substring(0,2), 16),
                         Integer.parseInt(hex.substring(2,4), 16),
                         Integer.parseInt(hex.substring(4,6), 16));
    }

    public static void createCard(String name, String message, int template,
                                  String bgHex, String textHex, String photoPath,
                                  String bgImagePath, String output) throws Exception {
        int width = 800, height = 600;
        BufferedImage img = new BufferedImage(width, height, BufferedImage.TYPE_INT_RGB);
        Graphics2D g = img.createGraphics();

        Color bgColor = bgHex != null ? hexToColor(bgHex) : TEMPLATES.get(template)[0];
        if (bgImagePath != null && new File(bgImagePath).exists()) {
            BufferedImage bgImg = ImageIO.read(new File(bgImagePath));
            g.drawImage(bgImg, 0, 0, width, height, null);
        } else {
            g.setColor(bgColor);
            g.fillRect(0, 0, width, height);
        }

        Color textColor = textHex != null ? hexToColor(textHex) : TEMPLATES.get(template)[1];
        g.setColor(textColor);
        g.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_ON);
        g.setFont(new Font("Liberation Sans", Font.BOLD, 60));
        FontMetrics fm = g.getFontMetrics();
        String title = "❄️ Happy New Year! ❄️";
        g.drawString(title, (width - fm.stringWidth(title))/2, 80);

        g.setFont(new Font("Liberation Sans", Font.BOLD, 70));
        fm = g.getFontMetrics();
        g.drawString(name, (width - fm.stringWidth(name))/2, 200);

        g.setFont(new Font("Liberation Sans", Font.ITALIC, 30));
        fm = g.getFontMetrics();
        String[] words = message.split(" ");
        StringBuilder line = new StringBuilder();
        java.util.List<String> lines = new ArrayList<>();
        for (String w : words) {
            if (line.length() + w.length() + 1 <= 30) {
                if (line.length() > 0) line.append(" ");
                line.append(w);
            } else {
                lines.add(line.toString());
                line = new StringBuilder(w);
            }
        }
        if (line.length() > 0) lines.add(line.toString());
        int y = 380;
        for (String l : lines) {
            g.drawString(l, (width - fm.stringWidth(l))/2, y);
            y += 40;
        }

        g.setFont(new Font("Liberation Sans", Font.PLAIN, 40));
        fm = g.getFontMetrics();
        String emoji = "⭐  🎄  🎆";
        g.drawString(emoji, (width - fm.stringWidth(emoji))/2, 520);

        if (photoPath != null && new File(photoPath).exists()) {
            BufferedImage photo = ImageIO.read(new File(photoPath));
            int size = 150;
            g.drawImage(photo, width - size - 20, 20, size, size, null);
        }

        g.dispose();
        ImageIO.write(img, "png", new File(output));
        System.out.println("Card saved to " + output);
    }

    public static void main(String[] args) throws Exception {
        Map<String, String> params = new HashMap<>();
        for (int i = 0; i < args.length; i++) {
            if (args[i].startsWith("-")) {
                String key = args[i].replaceFirst("^-+", "");
                if (i+1 < args.length && !args[i+1].startsWith("-")) {
                    params.put(key, args[++i]);
                } else {
                    params.put(key, "");
                }
            }
        }
        if (!params.containsKey("n")) {
            System.err.println("Error: -n name required");
            System.exit(1);
        }
        if (!params.containsKey("m")) {
            System.err.println("Error: -m message required");
            System.exit(1);
        }
        String name = params.get("n");
        String message = params.get("m");
        int template = params.containsKey("t") ? Integer.parseInt(params.get("t")) : 1;
        if (template < 1 || template > 3) {
            System.err.println("Template must be 1-3");
            System.exit(1);
        }
        String bg = params.get("bg");
        String text = params.get("text");
        String photo = params.get("photo");
        String bgImage = params.get("bg-image");
        String output = params.getOrDefault("o", "ny_card.png");

        createCard(name, message, template, bg, text, photo, bgImage, output);

        if (params.containsKey("html")) {
            String htmlFile = params.get("html");
            String bgColor = bg != null ? bg : "#FFFFFF";
            String textColor = text != null ? text : "#000000";
            String html = String.format("<!DOCTYPE html><html><head><title>New Year Card</title>" +
                    "<style>body{font-family:sans-serif;text-align:center;background:#1a1a2e;}" +
                    ".card{background:%s;color:%s;border-radius:20px;padding:40px;max-width:600px;margin:50px auto;box-shadow:0 4px 8px rgba(0,0,0,0.3);}" +
                    "h1{font-size:3em;}.name{font-size:2.5em;}.msg{font-size:1.5em;}</style>" +
                    "</head><body><div class=\"card\"><h1>❄️ Happy New Year! ❄️</h1>" +
                    "<div class=\"name\">%s</div><div class=\"msg\">%s</div><div style=\"font-size:3em;\">⭐ 🎄 🎆</div>" +
                    "</div></body></html>", bgColor, textColor, name, message);
            java.nio.file.Files.write(java.nio.file.Paths.get(htmlFile), html.getBytes());
            System.out.println("HTML card saved to " + htmlFile);
        }
    }
}

File: ny_card_creator.py
# ny_card_creator.py
import sys, os, argparse, json, textwrap
from PIL import Image, ImageDraw, ImageFont, ImageFilter

TEMPLATES = {
    1: {"name": "Classic", "bg": (255,255,255), "text": (0,0,0), "font_size": 60},
    2: {"name": "Modern", "bg": (0,100,150), "text": (255,215,0), "font_size": 70},
    3: {"name": "Festive", "bg": (200,50,50), "text": (255,255,255), "font_size": 65}
}

def create_card(name, message, template, bg_color, text_color, photo_path, bg_image_path, output):
    # Use template defaults if colors not overridden
    if bg_color is None:
        bg_color = TEMPLATES[template]["bg"]
    else:
        bg_color = tuple(int(bg_color.lstrip('#')[i:i+2], 16) for i in (0,2,4))
    if text_color is None:
        text_color = TEMPLATES[template]["text"]
    else:
        text_color = tuple(int(text_color.lstrip('#')[i:i+2], 16) for i in (0,2,4))

    width, height = 800, 600
    if bg_image_path and os.path.exists(bg_image_path):
        bg_img = Image.open(bg_image_path).resize((width, height))
        img = bg_img.convert("RGB")
    else:
        img = Image.new("RGB", (width, height), bg_color)

    draw = ImageDraw.Draw(img)
    try:
        font_large = ImageFont.truetype("/usr/share/fonts/truetype/liberation/LiberationSans-Bold.ttf", 60)
        font_medium = ImageFont.truetype("/usr/share/fonts/truetype/liberation/LiberationSans-Regular.ttf", 40)
        font_small = ImageFont.truetype("/usr/share/fonts/truetype/liberation/LiberationSans-Italic.ttf", 30)
    except:
        font_large = ImageFont.load_default()
        font_medium = ImageFont.load_default()
        font_small = ImageFont.load_default()

    # Title
    title = "❄️ Happy New Year! ❄️"
    draw.text((width//2, 50), title, font=font_large, fill=text_color, anchor="mt")

    # Name
    draw.text((width//2, 200), name, font=font_large, fill=text_color, anchor="mt")

    # Message with wrapping
    wrapped = textwrap.fill(message, width=20)
    draw.text((width//2, 380), wrapped, font=font_small, fill=text_color, anchor="mt")

    # Decorations
    draw.text((width//2, 500), "⭐  🎄  🎆", font=font_medium, fill=text_color, anchor="mt")

    # Photo
    if photo_path and os.path.exists(photo_path):
        try:
            photo = Image.open(photo_path)
            photo.thumbnail((150, 150))
            img.paste(photo, (width - photo.width - 20, 20))
        except Exception as e:
            print(f"Warning: could not embed photo: {e}", file=sys.stderr)

    img.save(output)
    print(f"Card saved to {output}")

def main():
    parser = argparse.ArgumentParser(description="New Year Card Creator")
    parser.add_argument('-n', '--name', required=True, help="Recipient's name")
    parser.add_argument('-m', '--message', required=True, help="Custom message")
    parser.add_argument('-t', '--template', type=int, choices=[1,2,3], default=1, help="Template (1-3)")
    parser.add_argument('--bg', help="Background color (hex, e.g. #FF0000)")
    parser.add_argument('--text', help="Text color (hex)")
    parser.add_argument('--photo', help="Path to photo to embed")
    parser.add_argument('--bg-image', help="Path to background image")
    parser.add_argument('-o', '--output', default="ny_card.png", help="Output PNG file")
    parser.add_argument('--html', help="Export to HTML file (optional)")
    args = parser.parse_args()

    create_card(args.name, args.message, args.template, args.bg, args.text, args.photo, args.bg_image, args.output)

    if args.html:
        bg = args.bg or "#FFFFFF"
        txt = args.text or "#000000"
        html_content = f"""<!DOCTYPE html>
<html><head><title>New Year Card</title>
<style>body{{font-family:sans-serif;text-align:center;background:#1a1a2e;}}
.card{{background:{bg};color:{txt};border-radius:20px;padding:40px;max-width:600px;margin:50px auto;box-shadow:0 4px 8px rgba(0,0,0,0.3);}}
h1{{font-size:3em;}} .name{{font-size:2.5em;}} .msg{{font-size:1.5em;}}</style>
</head><body><div class="card">
<h1>❄️ Happy New Year! ❄️</h1>
<div class="name">{args.name}</div>
<div class="msg">{args.message}</div>
<div style="font-size:3em;">⭐ 🎄 🎆</div>
</div></body></html>"""
        with open(args.html, 'w') as f:
            f.write(html_content)
        print(f"HTML card saved to {args.html}")

if __name__ == "__main__":
    main()
